vshard/storage/init.lua

```diff
@@ -1477,6 +1492,7 @@ local function bucket_recv_xc(bucket_id, from, data, opts)
             end
             limit = limit - 1
             if limit == 0 then
+                migration_notify(space.engine)
```

Comment:
That's a wrong place to put the notification ops: it should be placed right after the `box.begin()` so that it'll be the first operation in the transaction, not the last.

Reply:
Fixed as well as in all other cases

Reply:
I bet you're missing one at the line 1507 - when limit is reached we start another transaction.

Reply:
You're right. Fixed